dev-app/src/main/java/com/vincentmasselis/devapp/MainActivity.kt

```diff
@@ -16,9 +23,24 @@ class MainActivity : AppCompatActivity() {
 
         (getSystemService(Context.BLUETOOTH_SERVICE) as BluetoothManager)
             .rxScan()
-            .subscribe {
+            .subscribe ({
                 applicationContext
-            }
+            },
+                {
+                    when (it) {
+                        is NeedLocationPermission -> if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                            requestPermissions(arrayOf(Manifest.permission.ACCESS_FINE_LOCATION), 0)
+                        }
+                        is NeedBluetoothScanPermission -> if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+                            requestPermissions(arrayOf(Manifest.permission.BLUETOOTH_CONNECT, Manifest.permission.BLUETOOTH_SCAN), 0)
+                        }
+                        is NeedBluetoothConnectPermission -> if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+                            requestPermissions(arrayOf(Manifest.permission.BLUETOOTH_CONNECT, Manifest.permission.BLUETOOTH_SCAN), 0)
```

Comment:
The line #38 is the same than the line #35

Reply:
I tough it would be simplier to just ask for the 2 permissions at the same time to display only 1 popin to the user